src/main/java/Duke/TaskList.java

```diff
@@ -0,0 +1,169 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * A class that stores a list of tasks.
+ */
+public class TaskList {
+
+    private ArrayList<Task> list;
+    private int count;
+
+    /**
+     * Constructs an empty TaskList.
+     */
+    public TaskList() {
+        this.list = new ArrayList<>();
+        this.count = 0;
+    }
+
+    /**
+     * Constructs a TaskList with pre-existing Tasks from given Optional of a Stream of Strings.
+     * @param content  Optional of a Stream of Strings with supposed input of the save file.
```

Comment:
Maybe an empty line between description and param section? This applies to the other JavaDoc comments in other files.